src/encoded/reports/metadata.py

```diff
@@ -0,0 +1,380 @@
+    SEARCH_PATH = '/search/'
+    EXCLUDED_COLUMNS = (
+        'Restricted',
+        'No File Available',
+    )
+    DEFAULT_PARAMS = [
+        ('field', 'audit'),
+        ('field', 'files.@id'),
+        ('limit', 'all'),
+    ]
+
+    def __init__(self, request):
+        self.request = request
+        self.query_string = QueryString(request)
+        self.param_list = self.query_string.group_values_by_key()
+        self.positive_file_param_list = {}
+        self.header = []
+        self.experiment_column_to_fields_mapping = OrderedDict()
+        self.file_column_to_fields_mapping = OrderedDict()
+        self.visualizable_only = self.query_string.is_param('option', 'visualizable')
+        self.raw_only = self.query_string.is_param('option', 'raw')
+        self.search_request = None
+        self.csv = CSVGenerator()
+
+    def _get_column_to_fields_mapping(self):
+        return METADATA_COLUMN_TO_FIELDS_MAPPING
+
+    def _build_header(self):
+        for column in self._get_column_to_fields_mapping():
+            if column not in self.EXCLUDED_COLUMNS:
+                self.header.append(column)
+        for audit, column in METADATA_AUDIT_TO_AUDIT_COLUMN_MAPPING:
+            self.header.append(column)
+        
+    def _split_column_and_fields_by_experiment_and_file(self):
+        for column, fields in self._get_column_to_fields_mapping().items():
+            if fields[0].startswith('files'):
+                self.file_column_to_fields_mapping[column] = [
+                    field.replace('files.', '')
+                    for field in fields
+                ]
+            else:
+                self.experiment_column_to_fields_mapping[column] = fields
+
+    def _set_positive_file_param_list(self):
+        self.positive_file_param_list = {
+            k.replace('files.', ''): v
+            for k, v in self.param_list.items()
+            if k.startswith('files.') and '!' not in k
+        }
+
+    def _add_fields_to_param_list(self):
+        self.param_list['field'] = []
+        for column, fields in self._get_column_to_fields_mapping().items():
+            self.param_list['field'].extend(fields)
+
+    def _initialize_at_id_param(self):
+        self.param_list['@id'] = self.param_list.get('@id', [])
+
+    def _maybe_add_cart_elements_to_param_list(self):
+        cart_uuids = self.param_list.get('cart', [])
+        if cart_uuids:
+            try:
+                cart = self.request.embed(cart_uuids[0], '@@object')
+                del self.param_list['cart']
+            except KeyError:
+                raise HTTPBadRequest(explanation='Specified cart does not exist.')
+            else:
+                self.param_list['@id'].extend(
+                    cart.get('elements', [])
+                )
+
+    def _maybe_add_json_elements_to_param_list(self):
+        try:
+            self.param_list['@id'].extend(
+                self.request.json.get('elements', [])
+            )
+        except ValueError:
```

Comment:
This is lifted over from old code. I left it in because I was concerned the ValueError is coming from trying to access `.json` on `request` when it doesn't exist. Could probably clean this up.